clients/src/main/java/org/apache/kafka/common/utils/PoisonPill.java

```diff
@@ -50,7 +83,11 @@ public static void die(File heapDumpFolder, final long maxWaitForDump, int haltS
         }
     }
 
-    private static void grabHeapDump(File heapDumpFolder, final long maxWait, final int haltStatusCode) throws Exception {
+    public void recordTimeSinceLastDequeInMs(final long timeSinceLastDequeue) {
+        this.poisonPillStats.recordTimeSinceLastDequeInMs(timeSinceLastDequeue);
+    }
+
+    private void grabHeapDump(File heapDumpFolder, final long maxWait, final int haltStatusCode) throws Exception {
```

Comment:
Does it make a difference if the watch-dog thread is a daemon thread or not?

Reply:
According to the doc https://docs.oracle.com/javase/7/docs/api/java/lang/Thread.html
The Java Virtual Machine continues to execute threads until either of the following occurs:

The exit method of class Runtime has been called and the security manager has permitted the exit operation to take place.
All threads that are not daemon threads have died, either by returning from the call to the run method or by throwing an exception that propagates beyond the run method.


I think the intension of making the watcth-dog thread a daemon is that it does NOT block the shutdown.
However, since we'll eventually call Runtime.halt(), I'm not sure if it makes a real difference.
In any case, setting it to daemon is the safer way.